samtranslator/swagger/swagger.py

```diff
@@ -169,10 +169,18 @@ def add_lambda_integration(self, path, method, integration_uri,
         api_auth_config = api_auth_config or {}
         if method_auth_config.get('Authorizer') == 'AWS_IAM' \
            or api_auth_config.get('DefaultAuthorizer') == 'AWS_IAM' and not method_auth_config:
-            self.paths[path][method][self._X_APIGW_INTEGRATION]['credentials'] = self._generate_integration_credentials(
-                method_invoke_role=method_auth_config.get('InvokeRole'),
-                api_invoke_role=api_auth_config.get('InvokeRole')
+            method_invoke_role = method_auth_config.get('InvokeRole')
+            if not method_invoke_role and 'InvokeRole' in method_auth_config:
+                method_invoke_role = 'NONE'
+            api_invoke_role = api_auth_config.get('InvokeRole')
```

Comment:
This section looks for two things:
* If there is an `InvokeRole` set, it gets the value
* If `InvokeRole` is set to null, it sets it to 'NONE'
* If `InvokeRole` is not set, it is passed as usual and is set to `CallerCredentials`.